test/ask/survey_canceller_test.exs

```diff
@@ -234,23 +232,99 @@ defmodule Ask.SurveyCancellerTest do
       )
       |> Repo.insert()
     end
+  end
 
-    def wait_all_cancellations_from_pids(pids) do
-      pids
-      |> Enum.map(fn {_, pid} -> Process.monitor(pid) end)
-      |> Enum.each(&receive_down/1)
-    end
+  describe "failure resistance" do
+    test "keeps cancelling other respondents when cancelling some fail", %{user: user} do
+      project = create_project_for_user(user)
+      questionnaire = insert(:questionnaire, name: "test", project: project)
+      survey = cancelling_survey(project)
+      test_channel = TestChannel.new(false)
+
+      channel =
+        insert(:channel,
+          settings:
+            test_channel
+            |> TestChannel.settings(),
+          type: "sms"
+        )
+
+      respondent_group = create_group(survey, channel)
+      insert_list(5, :respondent, survey: survey, state: "active", respondent_group: respondent_group)
+
+      failing_respondent = insert(:respondent, survey: survey, state: "active", respondent_group: respondent_group)
+      failing_session = %Session{
+        current_mode: nil, # this will make the canceller fail
+        respondent: failing_respondent,
+        flow: %Flow{
+          questionnaire: questionnaire
+        },
+        schedule: survey.schedule
+      }
+
+      failing_respondent
+      |> Ask.Respondent.changeset(%{session: Session.dump(failing_session)})
+      |> Repo.update!()
 
-    def wait_all_cancellations_from_conn(conn) do
-      conn.assigns[:processors_pids]
-      |> Enum.map(&Process.monitor/1)
-      |> Enum.each(&receive_down/1)
+      insert_list(5, :respondent, survey: survey, state: "active", respondent_group: respondent_group)
+
+      simulate_survey_canceller_start()
+
+      wait_for_cancels(survey, 3)
```

Comment:
`3` is a _random_ number here, right? I mean, if the canceller is cancelling up to `100` respondents in a single call...
I would add a simple comment of this for the future devs, to prevent them (probably future you & myself) to rack their brains trying to figure out why `3` times